trin-metrics/src/utils.rs

```diff
@@ -0,0 +1,135 @@
+use std::time::Instant;
```

Comment:
*nitpick* this feels like more than a simple `util` imo, and deserves its own named module, something like `trin-metrics/src/timer.rs`